drivers/serial/uart_stm32.c

```diff
@@ -665,6 +671,13 @@ static int uart_stm32_init(struct device *dev)
 
 	LL_USART_Disable(UartInstance);
 
+#if defined(CONFIG_LPUART_1) \
+	&& defined(CONFIG_SOC_SERIES_STM32G4X) \
+	&& defined(CONFIG_UART_CONSOLE)
+	/* the clock source for LPUART console is set to HSI */
+	LL_APB1_GRP2_EnableClock(LL_APB1_GRP2_PERIPH_LPUART1);
+	LL_RCC_SetLPUARTClockSource(LL_RCC_LPUART1_CLKSOURCE_HSI);
```

Comment:
I'm a bit concerned: by doing it that way, we're introducing hard coded configuration which doesn't match default SoC configuration.
We should be able to use default source clck configuration, which doesn't require all this.
Can you elaborate why default PCLK source for LPUART doesn't work ?

Reply:
requires register access to g431 to investigate further